### PR TITLE
[SETL-1] README

### PR DESCRIPTION
**Pull request by hsnthn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,101 @@
...
 
-[![Join the chat at https://gitter.im/vngrs/spark-etl](https://badges.gitter.im/vngrs/spark-etl.svg)](https://gitter.im/vngrs/spark-etl?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
\ No newline at end of file
+[![Build Status](https://travis-ci.org/vngrs/spark-etl.svg?branch=master)](https://travis-ci.org/vngrs/spark-etl)
+[![Coverage Status](https://coveralls.io/repos/github/vngrs/spark-etl/badge.svg?branch=master)](https://coveralls.io/github/vngrs/spark-etl?branch=master)
+[![Join the chat at https://gitter.im/vngrs/spark-etl](https://badges.gitter.im/vngrs/spark-etl.svg)](https://gitter.im/vngrs/spark-etl?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+[The ETL(Extract-Transform-Load)] process is a key component of many data management operations, including move data and to transform the data from one format to another. To effectively support these operations, spark-etl is providig a distributed solution.
+
+spark-etl is a Scala based project and it is developing with Spark. So it is scalable and distributed. spark-etl will process data from N source to N database.
```
Comment (furkanvarol) [inline README.md]: from N sources to N destinations.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+  - Key Value Stores
+  - APIs
+  - Streams
+
+- Transform
+  - Row to json
+  - Row to csv
+  - Json to Row
+  - Change tables
+  - Merge tables
```
Comment (furkanvarol) [inline README.md]: records (instead of tables)

File: README.md [hunk 3]
```diff
@@ -1,3 +1,101 @@
...
+### Tech
+* [Scala] - Functional Programming Language
+* [ScalaTest] - ScalaTest is testing tool in the Scala ecosystem.
+* [wartremover] - WartRemover is a flexible Scala code linting tool.
+* [scalastyle] - Scalastyle examines Scala code and indicates potential problems with it.
+* [scoverage] - Scoverage is a code coverage tool for scala that offers statement and branch coverage. * [Apache Spark] - Apache Spark is a fast and general engine for large-scale data processing.
+* [travis-ci] - Travis CI is a hosted, distributed continuous integration service used to build and test software projects
+* [coveralls] - Coveralls is a web service to help you track your code coverage over time, and ensure that all your new code is fully covered
+
+And of course spark-etl itself is open source with a [public repository][etl]
```
Comment (furkanvarol) [inline README.md]: I think there is no need to keep this line since reader of this README file is already at our GitHub repo.

**Comment by furkanvarol** [comment]
@omerfarukdemir and @msimav, can you guys also check this?

File: README.md [hunk 4]
```diff
@@ -1,3 +1,101 @@
...
+
+### Todos
+
+ - Scalafmt integration
+ - ETL Design
+
+License
+----
+
+MIT License
```
Comment (omerfarukdemir) [inline README.md]: Add link to license file.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,101 @@
...
+  - Json to Row
+  - Change tables
+  - Merge tables
+
+- Load
+  - FILES (json, csv)
+  - SQLs
+  - NoSQLs
+  - Key Value Stores
+  - Elastic Search
```
Comment (omerfarukdemir) [inline README.md]: It is also one of NoSQLs. There is no reason to define it again here.

File: README.md [hunk 6]
```diff
@@ -1,3 +1,101 @@
...
+
+Being an active committer means participating on [gitter] discussions, helping to answer questions.
+Of course, contributing code and documentation to the project is important as well. A good way to start is contributing improvements, new features, or bug fixes. You need to show that you take responsibility for the code that you contribute, add tests and documentation, and help maintaining it.
+
+If you would like to become a committer, please write on [gitter].
+
+### Development
+
+spark-etl uses Scala and Apache Spark for distributed developing.
+The spark-etl committers are using IntelliJ IDEA to develop the spark-etl codebase. We recommend IntelliJ IDEA for developing projects that involve Scala code.
```
Comment (omerfarukdemir) [inline README.md]: We shouldn't recommend any IDE to developers. We don't depend on any of them.

**Comment by omerfarukdemir** [comment]
@furkanvarol I did and left some comments.

**Comment by furkanvarol** [comment]
Seems like this PR is ready merge. @omerfarukdemir and @msimav, do you want to add something?

**Comment by msimav** [comment]
:shipit:

File: README.md [hunk 7]
```diff
@@ -1,3 +1,98 @@
...
+
+- Load
+  - FILES (json, csv)
+  - SQLs
+  - NoSQLs
+  - Key Value Stores
+  - APIs
+  - Streams
+
+Differences between other ETL projects:
```
Comment (kubudi) [inline README.md:34]: I believe we should elaborate this part

File: README.md [hunk 8]
```diff
@@ -1,3 +1,98 @@
...
+  - Streams
+
+- Transform
+  - Row to json
+  - Row to csv
+  - Json to Row
+  - Change records
+  - Merge records
+
+- Load
```
Comment (kubudi) [inline README.md:26]: I know it's used as Load in spark but the term "load" leads me to think that we are loading somethin to our system, maybe we should explain it better.

Same applies for usage of this term in code as well

File: README.md [hunk 9]
```diff
@@ -1,3 +1,98 @@
...
+  - parallel ETL on cluster level
+  - synchronisation
+  - open source
+
+Example scenario to use spark-etl:
+We want to get data from multiple sources like MySQL and CVS. When we extracting data, we also want to filter and merge some fields/tables. During the transform layer, we want to run an SQL. Then we want to write the transformed data to multiple targets like S3 and Redshift.
+
+spark-etl is the easiest way to do this scenario!
+
+### Version
```
Comment (kubudi) [inline README.md:44]: This information is redundant, User can find the version from github interface.

File: README.md [hunk 10]
```diff
@@ -1,3 +1,98 @@
...
+  - APIs
+  - Streams
+
+Differences between other ETL projects:
+  - parallel ETL on cluster level
+  - synchronisation
+  - open source
+
+Example scenario to use spark-etl:
+We want to get data from multiple sources like MySQL and CVS. When we extracting data, we also want to filter and merge some fields/tables. During the transform layer, we want to run an SQL. Then we want to write the transformed data to multiple targets like S3 and Redshift.
```
Comment (kubudi) [inline README.md:40]: Maybe an illustration or a diagram can help users to understand the scenario

File: README.md [hunk 11]
```diff
@@ -1,3 +1,98 @@
...
+
+spark-etl is the easiest way to do this scenario!
+
+### Version
+0.0.1-SNAPSHOT
+
+### Tech
```
Comment (kubudi) [inline README.md:47]: I believe we should backlink to the technologies/libraries we use

File: README.md [hunk 12]
```diff
@@ -1,3 +1,98 @@
...
+
+Prerequisites for building spark-etl:
+
+- sbt clean assembly
+
+### How to become a committer
+
+Want to contribute? Great!
+
+Committers have write access to the project’s repositories, i.e., they can modify the code, documentation by themselves and also accept other contributions.
```
Comment (kubudi) [inline README.md:67]: Let's redirect possible committers to the gitter channel directly. No one wants to read this.
Maybe we can add a commonly known contribution guide link.

Note: Why do contributors have write access? I believe we should manage this by fork and prs

File: README.md [hunk 13]
```diff
@@ -1,3 +1,98 @@
...
+
+There is no strict protocol for becoming a committer.
+
+Being an active committer means participating on [gitter] discussions, helping to answer questions.
+Of course, contributing code and documentation to the project is important as well. A good way to start is contributing improvements, new features, or bug fixes. You need to show that you take responsibility for the code that you contribute, add tests and documentation, and help to maintain it.
+
+If you would like to become a committer, please write on [gitter].
+
+### Development
```
Comment (kubudi) [inline README.md:76]: We already said this on Tech section

**Comment by furkanvarol** [comment]
@hsnthn do you want to re-write the readme with @kubudi's suggests? If so please let me know and I'll revert this PR.

**Pull request merged**